bench/app.py

```diff
@@ -150,24 +150,36 @@ def switch_branch(branch, apps=None, bench='.', upgrade=False):
 	from .utils import update_requirements, backup_all_sites, patch_sites, build_assets, pre_upgrade, post_upgrade
 	import utils
 	apps_dir = os.path.join(bench, 'apps')
-	version_upgrade = is_version_upgrade(bench=bench, branch=branch)
-	if version_upgrade[0] and not upgrade:
-		raise MajorVersionUpgradeException("Switching to {0} will cause upgrade from {1} to {2}. Pass --upgrade to confirm".format(branch, version_upgrade[1], version_upgrade[2]), version_upgrade[1], version_upgrade[2])
 
 	if not apps:
-		apps = ['frappe', 'erpnext']
+		apps = [name for name in os.listdir(apps_dir)
+			if os.path.isdir(os.path.join(apps_dir, name))]
 		if branch=="v4.x.x":
 			apps.append('shopping_cart')
+	if 'frappe' in apps:
+		version_upgrade = is_version_upgrade(bench=bench, branch=branch)
+		if version_upgrade[0] and not upgrade:
+			raise MajorVersionUpgradeException("Switching to {0} will cause upgrade from {1} to {2}. Pass --upgrade to confirm".format(branch, version_upgrade[1], version_upgrade[2]), version_upgrade[1], version_upgrade[2])
+	else:
+		version_upgrade = (False,)
```

Comment:
Problem with that is that, we wouldn't know the git remote name of any custom app. We can do this for frappe-only apps though.